File: Leetcode_重刷紀錄/0012.Problems11. Container With Most Water/ContainerWithMostWater.cpp
//11. Container With Most Water
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0, right = height.size() - 1;
        int maxArea = 0,w;

        while ((w = right - left)) {
            if (height[left] < height[right]) {
                    maxArea = max(maxArea, height[left++] * w);   
                    } else {
                    maxArea = max(maxArea, height[right--] * w); 
                    }           
        }
        return maxArea;
    }
};

File: Leetcode_重刷紀錄/0012.Problems11. Container With Most Water/ContainerWithMostWater.java
//11. Container With Most Water
public class Solution {
    public int maxArea(int[] height) {
        int left = 0, right = height.length - 1;
        int maxArea = 0,w;
        while ((w = right - left)>0) {
            if (height[left] < height[right]) {
            maxArea = Math.max(maxArea, height[left++] * w);   
            } else {
            maxArea = Math.max(maxArea, height[right--] * w);  
            }
        }
        return maxArea;
    }
}

